components/admin/reports/GenerateReportModal.tsx

```diff
@@ -81,15 +106,24 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
   const handleSubmit = async () => {
     const columns = [];
 
-    userID && columns.push(ApplicationsReportColumn.UserId);
-    applicantName && columns.push(ApplicationsReportColumn.ApplicantName);
-    applicantDoB && columns.push(ApplicationsReportColumn.ApplicantDateOfBirth);
-    APPNumber && columns.push(ApplicationsReportColumn.AppNumber);
-    applicationDate && columns.push(ApplicationsReportColumn.ApplicationDate);
-    paymentMethod && columns.push(ApplicationsReportColumn.PaymentMethod);
-    feeAmount && columns.push(ApplicationsReportColumn.FeeAmount);
-    donationAmount && columns.push(ApplicationsReportColumn.DonationAmount);
-    totalAmount && columns.push(ApplicationsReportColumn.TotalAmount);
+    selectedColumns.has(ApplicationsReportColumn.UserId) &&
+      columns.push(ApplicationsReportColumn.UserId);
+    selectedColumns.has(ApplicationsReportColumn.ApplicantName) &&
+      columns.push(ApplicationsReportColumn.ApplicantName);
+    selectedColumns.has(ApplicationsReportColumn.ApplicantDateOfBirth) &&
+      columns.push(ApplicationsReportColumn.ApplicantDateOfBirth);
+    selectedColumns.has(ApplicationsReportColumn.AppNumber) &&
+      columns.push(ApplicationsReportColumn.AppNumber);
+    selectedColumns.has(ApplicationsReportColumn.ApplicationDate) &&
+      columns.push(ApplicationsReportColumn.ApplicationDate);
+    selectedColumns.has(ApplicationsReportColumn.PaymentMethod) &&
+      columns.push(ApplicationsReportColumn.PaymentMethod);
+    selectedColumns.has(ApplicationsReportColumn.FeeAmount) &&
+      columns.push(ApplicationsReportColumn.FeeAmount);
+    selectedColumns.has(ApplicationsReportColumn.DonationAmount) &&
+      columns.push(ApplicationsReportColumn.DonationAmount);
+    selectedColumns.has(ApplicationsReportColumn.TotalAmount) &&
+      columns.push(ApplicationsReportColumn.TotalAmount);
```

Comment:
Isn't the backend API order-agnostic? If that's the case, we can just convert the set to an array using the spread operator

Reply:
lol true, addressed!